crates/router/src/core/payments/operations/payment_attempt_record.rs

```diff
@@ -261,14 +263,29 @@ impl<F: Clone + Sync> UpdateTracker<F, PaymentAttemptRecordData<F>, PaymentsAtte
             common_enums::TriggeredBy::Internal => Some(payment_data.payment_attempt.id.clone()),
             common_enums::TriggeredBy::External => None,
         };
+        let active_attempts_group_id = payment_data.payment_attempt.attempts_group_id.clone();
+        let active_attempt_id_type = Some(common_enums::ActiveAttemptIDType::GroupID);
+        let amount_captured = payment_data.payment_intent.amount_captured;
+        let status = if amount_captured > Some(MinorUnit::new(0))
+            && *payment_data
+                .payment_intent
+                .enable_partial_authorization
+                .unwrap_or(false.into())
```

Comment:
If this is domain type, these types of inline unwrap shouldn't not happen, when the domain type is created at single place what is the default needs to populated and not in every single place where this data is used.